src/main/java/org/gridsuite/geodata/server/GeoDataController.java

```diff
@@ -69,13 +76,19 @@ public ResponseEntity<List<SubstationGeoData>> getSubstations(@Parameter(descrip
     @ApiResponses(value = {@ApiResponse(responseCode = "200", description = "Lines geographical data")})
     public ResponseEntity<List<LineGeoData>> getLines(@Parameter(description = "Network UUID")@RequestParam UUID networkUuid,
                                                       @Parameter(description = "Variant Id") @RequestParam(name = "variantId", required = false) String variantId,
-                                                      @Parameter(description = "Countries")@RequestParam(required = false) List<String> countries) {
+                                                      @Parameter(description = "Countries") @RequestParam(required = false) List<String> countries,
+                                                      @Parameter(description = "Lines ids") @RequestParam(name = "lineId", required = false) List<String> linesIds) {
         Set<Country> countrySet = toCountrySet(countries);
         Network network = networkStoreService.getNetwork(networkUuid);
         if (variantId != null) {
             network.getVariantManager().setWorkingVariant(variantId);
         }
-        List<LineGeoData> lines = geoDataService.getLines(network, countrySet);
+        List<LineGeoData> lines;
+        if (linesIds != null) {
+            lines = geoDataService.getLines(network, linesIds);
+        } else {
+            lines = geoDataService.getLines(network, countrySet);
```

Comment:
Same question than for the substations, we don't manage getLines with CountrySet AND linesIds ?